backend/api/auth/auth.service.js

```
@@ -1,29 +1,31 @@
 const bcrypt = require('bcrypt')
 const influencerService = require('../influencer/influencer.service.js')
 const brandService = require('../brand/brand.service.js')
+const userService = require('../user/user.service.js');
 const logger = require('../../services/logger.service')
 
 const saltRounds = 10
 
-async function login(email, password) {
-    logger.debug(`auth.service - login with email: ${email}`)
-    if (!email || !password) return Promise.reject('email and password are required!')
-
-    const user = await userService.getByEmail(email)
-    if (!user) return Promise.reject('Invalid email or password')
-    const match = await bcrypt.compare(password, user.password)
-    if (!match) return Promise.reject('Invalid email or password')
-
-    delete user.password;
+async function login(username, password, userType) {
+    
+    logger.debug(`auth.service - login with username: ${username}`)
+    if (!username || !password) return Promise.reject('username and password are required!')
+    
+    const user = await userService.getByUsername(username, userType)
+    if (!user) return Promise.reject('Invalid username or password')
```

if we use async / await syntax, let's " throw new Error" in case of errors, instead of returning Promise.reject.
in the outer function that called to this function, you should wrap the call with try-catch and react to relevant result.

---

done